cluster/src/main/java/io/atomix/cluster/protocol/SwimMembershipProtocol.java

```diff
@@ -239,9 +252,15 @@ else if (member.incarnationNumber() > swimMember.getIncarnationNumber()) {
 
         // If the state has been changed to ALIVE, trigger a REACHABILITY_CHANGED event and then update metadata.
         if (member.state() == State.ALIVE && swimMember.getState() != State.ALIVE) {
+          State swimState = swimMember.getState();
           swimMember.setState(State.ALIVE);
-          LOGGER.debug("{} - Member reachable {}", this.localMember.id(), swimMember);
-          post(new GroupMembershipEvent(GroupMembershipEvent.Type.REACHABILITY_CHANGED, swimMember.copy()));
+          if (config.isRetainTombstones() && swimState == State.DEAD) {
+            LOGGER.debug("{} - Member added {}", this.localMember.id(), swimMember);
+            post(new GroupMembershipEvent(GroupMembershipEvent.Type.MEMBER_ADDED, swimMember.copy()));
```

Comment:
MEMBER_ADDED when dead?

Reply:
Yeah, so we're seeing some weird timing issues when Atomix is run inside Kubernetes. Basically, some nodes are still getting `ALIVE` messages for a killed pod even after the 10 second window that's allowed before the member is marked `DEAD`. The default behavior was to remove the member from the state altogether, and that caused later messages to appear like valid states, causing nodes to repeatedly trigger `MEMBER_ADDED` and `MEMBER_REMOVED`. So this PR changes the default behavior to retain member states forever, ensuring nodes can correctly identify those errant `ALIVE` messages as out-of-date. But the member states that are retained as tombstones are hidden by the interface (see `getMembers`). To ensure the events are consistent with prior versions, when a member is marked `DEAD` a `MEMBER_REMOVED` event is still produced, and when a member is marked `ALIVE` a `MEMBER_ADDED` event is still produced. `DEAD` members can only be marked `ALIVE` once the associated member has learned it was marked `DEAD`, ensuring the node is actually running again.

Reply:
Easy to unit test?